Microsoft.Azure.Cosmos/src/Telemetry/TelemetryToServiceHelper.cs

```diff
@@ -86,15 +93,26 @@ private async Task RetrieveConfigAndInitiateTelemetryAsync()
             try
             {
                 Uri serviceEndpointWithPath = new Uri(this.serviceEnpoint + Paths.ClientConfigPathSegment);
-
-                TryCatch<AccountClientConfiguration> databaseAccountClientConfigs = await this.GetDatabaseAccountClientConfigAsync(this.cosmosAuthorization, this.httpClient, serviceEndpointWithPath);
-                if (databaseAccountClientConfigs.Succeeded)
-                {
-                    this.InitializeClientTelemetry(databaseAccountClientConfigs.Result);
-                }
-                else if (!this.cancellationTokenSource.IsCancellationRequested)
+                while (!this.cancellationTokenSource.IsCancellationRequested)
                 {
-                    DefaultTrace.TraceWarning($"Exception while calling client config " + databaseAccountClientConfigs.Exception.ToString());
+                    TryCatch<AccountClientConfiguration> databaseAccountClientConfigs = await this.GetDatabaseAccountClientConfigAsync(
+                        cosmosAuthorization: this.cosmosAuthorization,
+                        httpClient: this.httpClient, 
+                        clientConfigEndpoint: serviceEndpointWithPath);
+
+                    if (databaseAccountClientConfigs.Succeeded)
+                    {
+                        this.InitializeClientTelemetry(
+                            clientConfig: databaseAccountClientConfigs.Result);
+                    }
+                    else if (!this.cancellationTokenSource.IsCancellationRequested)
+                    {
+                        DefaultTrace.TraceWarning($"Exception while calling client config " + databaseAccountClientConfigs.Exception.ToString());
```

Comment:
Avoid string interpolation unless there is a trace listener:

```suggestion
                        DefaultTrace.TraceWarning("Exception while calling client config {0}", databaseAccountClientConfigs.Exception);
```

Reply:
Same thing in lines 102 below